internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,538 @@
+	stateCh := make(chan resolver.State, 1)
+	errCh := make(chan error, 1)
+
+	tcc := &testutils.ResolverClientConn{
+		Logger:       t,
+		UpdateStateF: func(s resolver.State) error { stateCh <- s; return nil },
+		ReportErrorF: func(err error) { errCh <- err },
+	}
+	return tcc, stateCh, errCh
+}
+
+// Tests the scenario where no proxy environment variables are set or proxying
+// is disabled by the `NO_PROXY` environment variable. The test verifies that
+// the delegating resolver creates only a target resolver and that the
+// addresses returned by the delegating resolver exactly match those returned
+// by the target resolver.
+func (s) TestDelegatingResolverNoProxyEnvVarsSet(t *testing.T) {
+	hpfe := func(req *http.Request) (*url.URL, error) { return nil, nil }
+	originalhpfe := delegatingresolver.HTTPSProxyFromEnvironment
+	delegatingresolver.HTTPSProxyFromEnvironment = hpfe
+	defer func() {
+		delegatingresolver.HTTPSProxyFromEnvironment = originalhpfe
+	}()
+
+	const (
+		targetTestAddr          = "test.com"
+		resolvedTargetTestAddr1 = "1.1.1.1:8080"
+		resolvedTargetTestAddr2 = "2.2.2.2:8080"
+	)
+	targetResolver := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := targetResolver.Scheme() + ":///" + targetTestAddr
+
+	// Create a delegating resolver with no proxy configuration
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	if _, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, targetResolver, false); err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	// Update the manual resolver with a test address.
+	targetResolver.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedTargetTestAddr1},
+			{Addr: resolvedTargetTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same addresses, as returned
+	// by the target resolver.
+	wantState := resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedTargetTestAddr1},
+			{Addr: resolvedTargetTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
+
+	var gotState resolver.State
+	select {
+	case gotState = <-stateCh:
+	case <-time.After(defaultTestTimeout):
+		t.Fatal("Timeout when waiting for a state update from the delegating resolver")
+	}
+
+	if diff := cmp.Diff(gotState, wantState); diff != "" {
+		t.Fatalf("Unexpected state from delegating resolver. Diff (-got +want):\n%v", diff)
+	}
+}
+
+// setupDNS registers a new manual resolver for the DNS scheme, effectively
+// overwriting the previously registered DNS resolver. This allows the test to
+// mock the DNS resolution for the proxy resolver. It also registers the
+// original DNS resolver after the test is done.
+func setupDNS(t *testing.T) *manual.Resolver {
+	mr := manual.NewBuilderWithScheme("dns")
```

Comment:
Call `t.Helper()` at the top to mark this is a test helper for better error reporting.

https://google.github.io/styleguide/go/decisions#test-helpers

Reply:
Done.